x-pack/plugins/observability_solution/synthetics/server/routes/monitor_cruds/delete_monitor.ts

```diff
@@ -29,30 +29,42 @@ import { formatSecrets, normalizeSecrets } from '../../synthetics_service/utils/
 
 export const deleteSyntheticsMonitorRoute: SyntheticsRestApiRouteFactory<
   DeleteParamsResponse[],
-  Record<string, any>,
-  Record<string, any>,
+  Record<string, string>,
+  Record<string, string>,
   { ids: string[] }
 > = () => ({
   method: 'DELETE',
-  path: SYNTHETICS_API_URLS.SYNTHETICS_MONITORS,
+  path: SYNTHETICS_API_URLS.SYNTHETICS_MONITORS + '/{id?}',
   validate: {},
   validation: {
     request: {
-      body: schema.object({
-        ids: schema.arrayOf(schema.string(), {
-          minSize: 1,
-        }),
+      body: schema.nullable(
+        schema.object({
+          ids: schema.arrayOf(schema.string(), {
+            minSize: 1,
+          }),
+        })
+      ),
+      params: schema.object({
+        id: schema.maybe(schema.string()),
       }),
     },
   },
   handler: async (routeContext): Promise<any> => {
     const { request, response } = routeContext;
 
-    const { ids } = request.body;
+    const { ids } = request.body || {};
+    const { id: queryId } = request.params;
 
     const result: Array<{ id: string; deleted: boolean; error?: string }> = [];
+    const idsToDelete = [...(ids ?? []), ...(queryId ? [queryId] : [])];
```

Comment:
I don't love this. This combines the request param and the body ids. I think we should return a 400 if they provide both. I wouldn't think that would be intended.

Reply:
done !!